examples/ui/relative_cursor_position.rs

```diff
@@ -13,7 +15,17 @@ fn main() {
 }
 
 fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
-    commands.spawn(Camera2dBundle::default());
+    commands.spawn(Camera2dBundle {
+        camera: Camera {
+            viewport: Some(Viewport {
+                physical_position: [200, 100].into(),
+                physical_size: [600, 600].into(),
+                ..default()
+            }),
```

Comment:
This breaks the example for me, and it may be related to scale factor (I will investigate)

Reply:
I fount the problem. The camera was rendering into a viewport, but UiCamera was not set on root UI node, and it defaulted to rendering into the primary window. This explains misalignment between cursor position and pixels displayed.

Setting viewport on a camera without linking it to the UI root node is an easy mistake to make, and I am not sure what would be an optimal API design to prevent it. At some point I used to default to the first camera rendering to the primary window if no UiCamera is set on a node, which would prevent this issue from happening. But then I faced a lot of tests failing without at least one camera existing in the world, and I switched the default from camera to the primary window.

I guess defaulting to a primary window *camera* (as I used to do) with a second fallback to the primary window itself (which I currently do) might be the way to do it.

I will investigate if this second fallback even makes sense, and if rendering UI to the primary window with no camera existing even worked before my changes.

Reply:
I wonder if it might make to not have a default at all: if you want to render UI then you need to set up a camera / viewport to render into.

> I guess defaulting to a primary window camera (as I used to do) with a second fallback to the primary window itself (which I currently do) might be the way to do it.

When you say "fallback to the primary window", do you mean that you can render UI to a window without going via a camera? I assumed that the way this would work would be that UI would always be camera driven and that "defaulting to the primary window" would just mean "defaulting to the primary camera of the primary window" (which I'm guessing is created automatically?)

Reply:
Other suggestion that I actually prefer:
- we separate UI cameras from 2d/3d cameras, into their own marker type `CameraUi`, remove `UiCameraConfig` (and probably rename `UiCamera`)
- the UI plugin automatically spawns a UI camera for the primary window, and marks it at the default UI camera
- UI root nodes that are not assigned a camera use the default one

Advantages:
- exactly same ergonomics as previously for simple cases
- better ergonomics if you have multiple 2d/3d cameras, since there is no need to hide the UI in `UiCameraConfig`s
- UI cameras actually have their own `Camera` component. They don't need to share it with an unrelated camera, so it can be set for its needs (`order`, `viewport`, `output_mode`, ...)
- the UI's `ExtractedView` component lives on the same (render world) entity as its `RenderPhase` (currently the `RenderPhase<TransparentUi>` is on the 2d/3d camera's entity instead). It matters because `RenderCommand`s can only access components on their item's entity and their view's entity. It's a small issue but I stumbled on it today.

Issue:

From what I remember, the main issue with separate UI cameras was that by default they don't clear the background before rendering (so that the 2d/3d scene is visible). So there is a problem if there are no 2d/3d cameras to clear it at the start of the frame. I don't think it's a difficult one to solve.

I'm aware that it's enough work to be a PR on its own, and probably controversial, but I wanted to write it somewhere and test the waters.

Reply:
> We could have a compromise between your two options: If only one camera exists, draw UI on it. If more than one camera exist, require all UI roots to select a camera (and warn if any UI root is not displayed because of that).

I like this idea. It will allow for a quick start, further customization, and will not require a significant rethinking of the rest of the work accomplished in this PR. I will implement it and get back with the results.

Reply:
Implemented in https://github.com/bevyengine/bevy/pull/10559/commits/402109beba0c3179df0a0af9b27c309b3a406cc4